contracts/token/GovernorMills.sol

```diff
@@ -0,0 +1,383 @@
+        mapping (address => Receipt) receipts;
+    }
+
+    /// @notice Ballot receipt record for a voter
+    struct Receipt {
+        /// @notice Whether or not a vote has been cast
+        bool hasVoted;
+
+        /// @notice Whether or not the voter supports the proposal
+        bool support;
+
+        /// @notice The number of votes the voter had, which were cast
+        uint96 votes;
+    }
+
+    /// @notice Possible states that a proposal may be in
+    enum ProposalState {
+        Pending,
+        Active,
+        Canceled,
+        Defeated,
+        Succeeded,
+        Queued,
+        Expired,
+        Executed
+    }
+
+    /// @notice The official record of all proposals ever proposed
+    mapping (uint => Proposal) public proposals;
+
+    /// @notice The latest proposal for each proposer
+    mapping (address => uint) public latestProposalIds;
+
+    /// @notice The EIP-712 typehash for the contract's domain
+    bytes32 public constant DOMAIN_TYPEHASH = keccak256("EIP712Domain(string name,uint256 chainId,address verifyingContract)");
+
+    /// @notice The EIP-712 typehash for the ballot struct used by the contract
+    bytes32 public constant BALLOT_TYPEHASH = keccak256("Ballot(uint256 proposalId,bool support)");
+
+    /// @notice An event emitted when a new proposal is created
+    event ProposalCreated(uint id, address proposer, address[] targets, uint[] values, string[] signatures, bytes[] calldatas, uint startBlock, uint endBlock, string description);
+
+    /// @notice An event emitted when a vote has been cast on a proposal
+    event VoteCast(address voter, uint proposalId, bool support, uint votes);
+
+    /// @notice An event emitted when a proposal has been canceled
+    event ProposalCanceled(uint id);
+
+    /// @notice An event emitted when a proposal has been queued in the Timelock
+    event ProposalQueued(uint id, uint eta);
+
+    /// @notice An event emitted when a proposal has been executed in the Timelock
+    event ProposalExecuted(uint id);
+
+    /// @notice An event emitted when a new guardian has been set
+    event NewGuardian(address guardian);
+
+    /// @notice An event emitted when proposal threshold is updated
+    event ProposalThresholdUpdated(uint256 oldThreshold, uint256 newThreshold);
+
+    constructor(address _timelock, address _inv, address _xinv, address _guardian, uint256 _threshold) public {
+        timelock = TimelockInterface(_timelock);
+        inv = InvInterface(_inv);
+        xinv = InvInterface(_xinv);
+        guardian = _guardian;
+        threshold = _threshold;
+    }
+
+    function _getPriorVotes(address _proposer, uint256 _blockNumber) internal view returns (uint256) {
+        uint256 invPriorVotes = uint256(inv.getPriorVotes(_proposer, _blockNumber));
+        uint256 xinvPriorVotes = uint256(xinv.getPriorVotes(_proposer, _blockNumber));
+        
+        return add256(invPriorVotes, xinvPriorVotes);
```

Comment:
1 xINV vote != 1 INV vote

We need to use the xINV exchange rate to convert xINV votes to INV